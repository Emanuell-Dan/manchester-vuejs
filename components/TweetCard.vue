<template>
  <ul 
    :class="[
      `tweet-card--${gridPosition}`,
      {
        'md:mt-16': gridPosition === 'left',
        'md:mt-32': gridPosition === 'right'
      }
    ]"
    class="tweet-card__list flex flex-col items-center mx-2 lg:mx-4">
    <li 
      v-for="(tweet, index) in tweets"
      :key="index"
      class="tweet-card__item bg-white flex items-center max-w-md rounded-lg my-2">
      <a 
        :href="`https://twitter.com/${getUserScreenName(index)}/status/${getUserId(index)}`"
        target="_blank"
        class="block py-4 px-6">
        <p 
          v-if="tweet.retweeted_status"
          class="tweet-card__retweeted relative pl-8 mb-2">
          Retweeted
        </p>
        <div class="tweet-card__header flex items-center mb-2">
          <img 
            :src="getProfileImage(index)" 
            alt=""
            class="tweet-card__profile-logo rounded">
          <p class="tweet-card__profile-name pl-2">{{ getUserName(index) }}</p>
        </div>

        <div class="tweet-card__content mb-2">
          <p class="tweet-card__text">{{ getTweetMessage(index) }}</p>
        </div>

        <div class="tweet-card__footer flex">
          <p class="tweet-card__count relative pl-8 pr-2 mr-2">{{ getRetweetsCount(index) }}</p>
          <p class="tweet-card__likes relative pl-8">{{ getLikesCount(index) }}</p>
        </div>
      </a>
    </li>
  </ul>
</template>

<script>
export default {
	name: 'TweetCard',
	props: {
		tweets: {
      type: Array,
      default: () => []
		},
		gridPosition: {
			type: String,
			required: true
		}
	},
	methods: {
		getUserScreenName(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.user.screen_name : this.tweets[index].user.screen_name;
		},
		getUserId(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.id_str : this.tweets[index].id_str;
		},
		getProfileImage(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.user.profile_image_url_https : this.tweets[index].user.profile_image_url_https;
		},
		getUserName(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.user.name : this.tweets[index].user.name;
		},
		getTweetMessage(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.text : this.tweets[index].text;
		},
		getRetweetsCount(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.retweet_count : this.tweets[index].retweet_count;
		},
		getLikesCount(index) {
			return this.tweets[index].retweeted_status ? this.tweets[index].retweeted_status.favorite_count : this.tweets[index].favorite_count;
		}
	}
}
</script>
