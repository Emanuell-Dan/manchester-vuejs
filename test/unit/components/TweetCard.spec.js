import TweetCard from '@/components/TweetCard.vue';
import { mount } from '@vue/test-utils';

describe('TweetCard.vue', () => {
	let cmp;

	beforeEach(() => {
		cmp = mount(TweetCard, {
			propsData: {
				tweets: [
					{
						user: {
							screen_name: 'Gill2929',
							name: 'Harry Hill',
						},
						text: 'twitter text here',
						retweet_count: 5,
						favorite_count: 15
					}
				]
			}
		});
	});

	it('should render with expected html structure', () => {
		expect(cmp.element).toMatchSnapshot();
	});

	describe('twitter card', () => {
		it('renders user name', () => {
			expect(cmp.find('.tweet-card__profile-name').text()).toBe('Harry Hill');
		});

		it('renders tweet text', () => {
			expect(cmp.find('.tweet-card__text').text()).toBe('twitter text here');
		});

		it('renders retweet count', () => {
			expect(cmp.find('.tweet-card__count').text()).toBe('5');
		});

		it('renders likes count', () => {
			expect(cmp.find('.tweet-card__likes').text()).toBe('15');
		});
	});
});
