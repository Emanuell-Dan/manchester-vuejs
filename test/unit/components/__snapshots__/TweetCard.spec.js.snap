// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetCard.vue should render with expected html structure 1`] = `
<ul
  class="tweet-card__list flex flex-col items-center mx-2 lg:mx-4 tweet-card--undefined"
>
  <li
    class="tweet-card__item bg-white flex items-center max-w-md rounded-lg my-2"
  >
    <a
      class="block py-4 px-6"
      href="https://twitter.com/Gill2929/status/undefined"
      target="_blank"
    >
      <!---->
       
      <div
        class="tweet-card__header flex items-center mb-2"
      >
        <img
          alt=""
          class="tweet-card__profile-logo rounded"
        />
         
        <p
          class="tweet-card__profile-name pl-2"
        >
          Harry Hill
        </p>
      </div>
       
      <div
        class="tweet-card__content mb-2"
      >
        <p
          class="tweet-card__text"
        >
          twitter text here
        </p>
      </div>
       
      <div
        class="tweet-card__footer flex"
      >
        <p
          class="tweet-card__count relative pl-8 pr-2 mr-2"
        >
          5
        </p>
         
        <p
          class="tweet-card__likes relative pl-8"
        >
          15
        </p>
      </div>
    </a>
  </li>
</ul>
`;
